cranelift/filetests/filetests/isa/x64/simd-lane-access-compile.clif

```diff
@@ -91,3 +91,34 @@ block0(v0: f64):
 ; check:  uninit  %xmm1
 ; nextln: movsd   %xmm0, %xmm1
 ; nextln: movlhps %xmm0, %xmm1
+
+
+
+;; load*_zero
+
+; Verify that a `load` followed by a `scalar_to_vector` (the CLIF translation of `load32_zero`) is
+; lowered to a single MOVSS instruction.
+function %load32_zero_coalesced(i64) -> i32x4 {
+block0(v0: i64):
+    v1 = load.i32 v0
+    v2 = scalar_to_vector.i32x4 v1
+    ; check:  movss   0(%rdi), %xmm0
+    return v2
```

Comment:
Did you work out what the issue was with the earlier version of this test that could not coalesce?

Reply:
I debug-ed to [here](https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/codegen/src/machinst/lower.rs#L918-L924) and realized that indeed the insertion of the block was preventing `input_to_reg_mem` from seeing the input as mergeable. I realized I didn't understand how the coloring is working, but my hazy mental model makes me think that the logic should not be "allow sinking if the color is +1" but rather "allow sinking if the color is 0 or +1". I mean, I haven't looked too closely at the logic, but from the comments in `src/machinst/lower.rs`, I would expect the following code to coalesce, but it does not:
```
function %load32_zero(i64) -> i32x4 {
block0(v0: i64):
    v1 = load.i32 v0
    v2 = scalar_to_vector.i32x4 v1
    jump block1
 block1:
    return v2
}
```

In any case, the optimization here is a separate issue than implementing the load*_zero instruction itself.

Reply:
So the reason for the +1 correction is that the test is comparing the *input* color of the possibly-merged load to the color at the merging point; and because loads are effectful (in this model), the *output* color of the load is input+1. So the test is really just "output color of load is current color". That said, it's possible that the new-block increment is happening at the wrong place such that we disallow this case -- can debug further in a bit...

Reply:
(To clarify the above: I'm fine with this PR merging as-is, but please feel free to create a tracking issue for this coalescing case if still unsolved!)

Reply:
Ok, opened #2562 to track this; will merge as-is to keep things moving along.